lib/client.js

```diff
@@ -18,70 +18,110 @@ var GiantSwarm = (function () {
 
   return {
 
-    _authorizedRequest: function() {
-      if (authToken) {
-        return request.set('Authorization', 'giantswarm ' + authToken);
+    // to be thrown in case of bad parameters/arguments/values given
+    UserException: function(message) {
+      this.message = message;
+      this.name = "UserException";
+      this.toString = function(){
+        return this.name + ": " + this.message;
+      };
+    },
+
+    // simplify validation of string parameters for internal methods
+    _validateStringParameters: function(parameters) {
+      for (name in parameters) {
+        if (!parameters[name]) {
+          throw new this.UserException("Parameter '"+ name +"' must be given");
+        }
+        if (typeof(parameters[name]) !== 'string') {
+          throw new this.UserException("Parameter '"+ name +"' must be of type string");
+        }
       }
     },
 
-    // allows the user to overwrite the default endpoint
+    /**
+     * Allows the user to overwrite the default API endpoint
+     * to be used by subsequent API calls
+     *
+     * @param {String} [url] The URL to use as endpoint, without trailing slash
+     */
     setApiEndpoint: function(url) {
+      this._validateStringParameters({url: url});
       apiEndpoint = url;
-      wsurl = url.replace('http://', 'ws://');
+      var wsurl = url.replace('http://', 'ws://');
       wsurl = wsurl.replace('https://', 'wss://');
       websocketEndpoint = wsurl;
     },
 
     /**
-     * successCallback: Called in case of success, no arguments
-     * errorCallback (optional): Called in case of an error, error object as argument
+     * Check the availability of the API
+     *
+     * @param {Function} [successCallback] Callback to be called in case of success, no arguments
+     * @param {Function} [errorCallback] Callback to be called in case of an error, error object as argument
      */
     ping: function(successCallback, errorCallback) {
+      if (typeof(successCallback) !== 'function') throw new this.UserException('successCallback must be given');
       request
         .get(apiEndpoint + '/v1/ping')
         .end(function(err, res){
-          if (res.text.trim() === '"OK"') {
-            successCallback();
-          } else if (errorCallback) {
-            errorCallback(err);
+          if (err) {
+            if (errorCallback) errorCallback(err);
```

Comment:
When error callback is a number that will fail. Lets use the same check as for the success callback.